swarm/src/lib.rs

```diff
@@ -1373,7 +1365,7 @@ where
             pool: Pool::new(self.local_peer_id, pool_config, self.connection_limits),
             behaviour: self.behaviour,
             supported_protocols,
-            listened_addrs: HashMap::new(),
+            listened_addrs: SmallVec::new(),
```

Comment:
Is stack- instead of heap-allocation necessary here? In other words, is this optimization worth it? Would a `Vec` suffice?

Reply:
That's how it is currently on master. What's the drawback of using stack-allocation / `SmallVec`?

Reply:
> That's how it is currently on master. 

Missed that. Sorry. Let's keep it as is then.

> What's the drawback of using stack-allocation / `SmallVec`?

No drawback. Though I am not aware of a proof that this is not a premature optimization.